src/librustc_typeck/check/op.rs

```diff
@@ -458,18 +485,27 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                         .is_ok()
                                 } {
                                     if let Ok(lstring) = source_map.span_to_snippet(lhs_expr.span) {
-                                        err.help(&format!(
-                                            "`{}` can be used on '{}', you can \
-                                            dereference `{2}`: `*{2}`",
-                                            op.node.as_str(),
-                                            rty.peel_refs(),
-                                            lstring
-                                        ));
+                                        err.span_suggestion_verbose(
```

Comment:
A lot of this error reporting seems duplicated between the `IsAssign::No` and `IsAssign::Yes` arms - is there something I'm missing or could we move some of this into a function and just call it from both arms?